core/src/main/java/org/apache/spark/unsafe/map/BytesToBytesMap.java

```diff
@@ -256,6 +272,15 @@ public boolean hasNext() {
     @Override
     public Location next() {
       int totalLength = PlatformDependent.UNSAFE.getInt(pageBaseObject, offsetInPage);
+      if (destructive) {
+        MemoryLocation keyAddress = loc.getKeyAddress();
+        Object keyBaseObject = keyAddress.getBaseObject();
+        long keyBaseOffset = keyAddress.getBaseOffset();
+
+        int hashcode = HASHER.hashUnsafeWords(keyBaseObject, keyBaseOffset, loc.getKeyLength());
+        int pos = hashcode & map.mask;
+        this.map.bitset.unset(pos);
+      }
```

Comment:
This block is what originally confused me.  This is going to involve a huge performance hit to have to re-hash each record. Do we need to free the bitset? I suppose that not doing so leaves things in a slightly inconsistent state, but in my imagined use case for the destructive iterator we would never call any methods on the map after calling the destructive iterator and thus this would be safe.  Is there some consideration that I've overlooked here?